src/plugins/auto/src/auto_compiled_model.cpp

```diff
@@ -108,50 +108,76 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
             real = m_scheduler->m_compile_context[ACTUALDEVICE].
                 m_compiled_model->get_property(name).as<unsigned int>();
         } else {
-            OPENVINO_ASSERT(m_scheduler->m_compile_context[CPU].m_is_already == true);
             std::unique_lock<std::mutex> lock(m_context->m_mutex);
             auto device_info = m_scheduler->m_compile_context[ACTUALDEVICE].m_device_info;
             lock.unlock();
             unsigned int optimal_batch_size = 0;
             unsigned int requests = 0;
             bool tput_enabled_in_plugin = false;
-            auto actual_dev_supported_properties = m_context->m_ov_core->get_property(device_info.device_name, ov::supported_properties);
+            auto actual_dev_supported_properties =
+                m_context->m_ov_core->get_property(device_info.device_name, ov::supported_properties);
+            bool is_supported_num_request = std::find(actual_dev_supported_properties.begin(),
+                                                      actual_dev_supported_properties.end(),
+                                                      ov::hint::num_requests) != actual_dev_supported_properties.end();
+            auto reqs_iter = device_info.config.find(ov::hint::num_requests.name());
+            auto ireq_iter = std::find(actual_dev_supported_properties.begin(), actual_dev_supported_properties.end(), name);
+            if (ireq_iter != actual_dev_supported_properties.end()) {
+                real = m_context->m_ov_core->get_property(device_info.device_name,
+                                                          ov::optimal_number_of_infer_requests,
+                                                          device_info.config);
+            }
+            if (real > 0) {
+                if (is_supported_num_request && reqs_iter != device_info.config.end()) {
+                    requests = reqs_iter->second.as<unsigned int>();
+                    real = (std::min)(requests, real);
```

Comment:
```suggestion
                    if (requests > 0)
                        real = (std::min)(requests, real);
```

Reply:
just want to make sure that the requests num here is not some default value carried from auto

Reply:
Sorry, didn't understand clearly what you meant. Could you please help provide a test case for this scenario?